### PR TITLE
Launching and funding the contest

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #7402 

## Description of Changes
- wired up FE to protocol helpers and platform api to be able to create contest and fund it

### Prerequisites 
- you need to have metamask with some sepolia-base coins on it
- Add sepolia-base to metamask (https://revoke.cash/learn/wallets/add-network/base-sepolia)
- Top up your wallet with some sepolia-base ETH with available faucets (eg https://cloud.google.com/application/web3/faucet/ethereum/sepolia)
- add FLAG_CONTEST=true to .env

### Creating community
- create new evm type community and pick “base sepolia” from “select chain input”
- enable stake within this community

https://github.com/hicommonwealth/commonwealth/assets/14819225/f7f903b5-1bb5-4416-8744-1173ddc1316a

### Launching contest
- as an admin of newly created community, go to contests in left sidebar
- because there are no contests yet, the empty page should appear with `launch contest` cta
- play around with a form, there are validation applied that would not let you go to the next step if something is wrong
  - you can create either recurring or one-off contest
  - for one-off, you need to specify funding token eg `0x0000000000000000000000000000000000000000`
-  in the next step, sign transaction using metamask
  - under the hood there are two transactions 
    - 1. contest creation onchain
    - 2. contest creation on platofrm side

https://github.com/hicommonwealth/commonwealth/assets/14819225/66cac639-e29d-4ee4-8bcf-2e12fbaab42a

### Funding contest
- right after contest is created you can fund it by opening the drawer
- funding form has some validation so play around if everything looks good
- after clicking `fund` cta, you need to sign metamask transaction
- after successfully funded contest, you can see the transaction on the etherscan by clicking `view transaction`
- the other way to fund contest is by going to `/contests` page where there is a list of contests and from there, clicking `fund` on contest card will open the funding drawer

https://github.com/hicommonwealth/commonwealth/assets/14819225/0d9129a2-9fc9-40fe-b500-3b47c5ce0806

## Deployment Plan

1. this is behind the feature flag FLAG_CONTEST so it is safe to deploy

## Other Considerations

- you can see the list of created contests in `/contests` route, however, there is gonna be another PR specifically for handling editing, canceling and displaying all the values regarding to contests data. Now these functionalities do not work properly

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/component_kit/CWContentPage/CWContentPage.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/component_kit/CWContentPage/CWContentPage.tsx:225]: this will be uncommented in upcoming pr

File: packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.tsx
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/ThreadCard.tsx:105]: this will be uncommented in upcoming pr

File: packages/commonwealth/client/scripts/helpers/ContractHelpers/Contest.ts
Comment (mzparacha) [inline packages/commonwealth/client/scripts/helpers/ContractHelpers/Contest.ts:68]: lets use a const for this

**Review by mzparacha — APPROVED** [review]

**Comment by ForestMars** [comment]
Should be able to get into `v1.4.0` on 5/17 assuming Marcin's catches this one conflict in the morning.

File: packages/commonwealth/client/scripts/helpers/ContractHelpers/Contest.ts
Comment (masvelio) [reply packages/commonwealth/client/scripts/helpers/ContractHelpers/Contest.ts:68]: done

**Comment by masvelio** [comment]
> Should be able to get into `v1.4.0` on 5/17 assuming Marcin's catches this one conflict in the morning.

fixed, still waiting for additional approval

**Review by burtonator — APPROVED** [review]

**Review by ianrowan — APPROVED** [review]
LGTM

**Pull request merged**
